pallets/transaction-payment-free/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+    decl_module,
+    weights::{DispatchInfo, PostDispatchInfo}
+};
+use sp_std::prelude::*;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use sp_runtime::{
+    traits::{DispatchInfoOf, Dispatchable, SignedExtension, Zero},
+    transaction_validity::TransactionValidityError,
+    FixedPointOperand
+};
+
+mod payment;
+
+pub use payment::*;
+
+type BalanceOf<T> = <<T as Config>::OnChargeTransaction as OnChargeTransaction<T>>::Balance;
+
+pub trait Config: frame_system::Config {
+    type OnChargeTransaction: OnChargeTransaction<Self>;
+}
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: T::Origin{}
+}
+
+/// Require the transactor have balance. All transactions are free - they have no fee
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct ChargeTransactionPayment<T: Config>(#[codec(compact)] BalanceOf<T>);
```

Comment:
I think we should rename this struct. `ChargeTransactionPayment` doesn't feel like it matches what this does